util/reflect_test.go

```diff
@@ -847,3 +958,399 @@ func TestStructValueHasNFields(t *testing.T) {
+}
+type KeyStruct2 struct {
+	Key1 string
+	Key2 int32
+}
+type ListElemStruct2 struct {
+	Key1  *string              `path:"key1"`
+	Key2  *int32               `path:"key2"`
+	Outer *OuterContainerType2 `path:"outer"`
+}
+type ContainerStruct2 struct {
+	StructKeyList map[KeyStruct2]*ListElemStruct2 `path:"struct-key-list"`
+}
+
+func (*InnerContainerType2) IsYANGGoStruct() {}
+func (*OuterContainerType2) IsYANGGoStruct() {}
+func (*ListElemStruct2) IsYANGGoStruct()     {}
+func (*ContainerStruct2) IsYANGGoStruct()    {}
+
+func TestGetNodesStructKeyedList(t *testing.T) {
+	containerWithLeafListSchema := &yang.Entry{
+		Name: "container",
+		Kind: yang.DirectoryEntry,
+		Dir: map[string]*yang.Entry{
+			"struct-key-list": {
+				Name:     "struct-key-list",
+				Kind:     yang.DirectoryEntry,
+				ListAttr: &yang.ListAttr{MinElements: &yang.Value{Name: "0"}},
+				Key:      "key1 key2",
+				Config:   yang.TSTrue,
+				Dir: map[string]*yang.Entry{
+					"key1": {
+						Name: "key1",
+						Kind: yang.LeafEntry,
+						Type: &yang.YangType{Kind: yang.Ystring},
+					},
+					"key2": {
+						Name: "key2",
+						Kind: yang.LeafEntry,
+						Type: &yang.YangType{Kind: yang.Yint32},
+					},
+					"outer": {
+						Name: "outer",
+						Kind: yang.DirectoryEntry,
+						Dir: map[string]*yang.Entry{
+							"inner": {
+								Name: "inner",
+								Kind: yang.DirectoryEntry,
+								Dir: map[string]*yang.Entry{
+									"leaf-field": {
+										Name: "leaf-field",
+										Kind: yang.LeafEntry,
+										Type: &yang.YangType{Kind: yang.Yint32},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	c1 := &ContainerStruct2{
+		StructKeyList: map[KeyStruct2]*ListElemStruct2{
+			{"forty-two", 42}: {
+				Key1:  String("forty-two"),
+				Key2:  Int32(42),
+				Outer: &OuterContainerType2{Inner: &InnerContainerType2{LeafName: Int32(1234)}},
+			},
+		},
+	}
+
+	tests := []struct {
+		desc       string
+		rootStruct interface{}
+		path       *gpb.Path
+		want       interface{}
+		wantErr    string
+	}{
+		{
+			desc:       "success leaf",
+			rootStruct: c1,
+			path: &gpb.Path{
+				Elem: []*gpb.PathElem{
+					{
+						Name: "struct-key-list",
+						Key: map[string]string{
+							"key1": "forty-two",
+							"key2": "42",
+						},
+					},
+					{
+						Name: "outer",
+					},
+					{
+						Name: "inner",
+					},
+					{
+						Name: "leaf-field",
+					},
+				},
+			},
+			want: []interface{}{c1.StructKeyList[KeyStruct2{"forty-two", 42}].Outer.Inner.LeafName},
+		},
+		{
+			desc:       "success container",
+			rootStruct: c1,
+			path: &gpb.Path{
+				Elem: []*gpb.PathElem{
+					{
+						Name: "struct-key-list",
+						Key: map[string]string{
+							"key1": "forty-two",
+							"key2": "42",
+						},
+					},
+					{
+						Name: "outer",
+					},
+					{
+						Name: "inner",
+					},
+				},
+			},
+			want: []interface{}{c1.StructKeyList[KeyStruct2{"forty-two", 42}].Outer.Inner},
+		},
+	}
```

Comment:
Would be good to test some of the failure cases here -- i.e., mismatched keys within the list, partial key matches, leaves that do not exist.

Reply:
Done.